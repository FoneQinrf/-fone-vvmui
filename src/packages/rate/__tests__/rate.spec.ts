/*
 * @Author: Fone`峰
 * @Date: 2021-06-09 11:47:19
 * @LastEditors: Fone`峰
 * @LastEditTime: 2021-06-09 13:24:29
 * @Description: file content
 * @Email: qinrifeng@163.com
 * @Github: https://github.com/FoneQinrf
 */
import { mount } from '@vue/test-utils';
import Rate from '../index.vue';
import { nextTick } from 'vue';

describe('Rate', () => {
  it('should count render corretly', async () => {
    const wrapper = mount(Rate);
    wrapper.setProps({ count: 7 });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findAll('.vvm-rate__icon').length === 7).toBe(true);
  });

  it('should disabled click corretly', async () => {
    const wrapper = mount(Rate, { props: { disabled: true } });
    await nextTick();
    wrapper.vm.change(1);
    await nextTick();
    expect(wrapper.vm.currentValue !== 1).toBe(true);
  });

  it('should click and emit corretly', async () => {
    const wrapper = mount(Rate);
    wrapper.vm.change(1);
    await nextTick();
    expect(wrapper.emitted()).toHaveProperty('change');
  });
});
