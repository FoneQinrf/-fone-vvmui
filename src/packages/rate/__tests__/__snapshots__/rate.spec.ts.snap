// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate should count render corretly 1`] = `
"<div class=\\"vvm-rate\\">
  <div class=\\"vvm-rate__icon\\" style=\\"width: 26px; height: 26px;\\">
    <transition-stub>
      <!--v-if-->
    </transition-stub><label class=\\"vvm-rate__defaultIcon\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconpingfenxingxing\\" style=\\"font-size: 26px; width: 26px; height: 26px; color: rgb(224, 224, 235);\\"></i></label>
  </div>
  <div class=\\"vvm-rate__icon\\" style=\\"width: 26px; height: 26px;\\">
    <transition-stub>
      <!--v-if-->
    </transition-stub><label class=\\"vvm-rate__defaultIcon\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconpingfenxingxing\\" style=\\"font-size: 26px; width: 26px; height: 26px; color: rgb(224, 224, 235);\\"></i></label>
  </div>
  <div class=\\"vvm-rate__icon\\" style=\\"width: 26px; height: 26px;\\">
    <transition-stub>
      <!--v-if-->
    </transition-stub><label class=\\"vvm-rate__defaultIcon\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconpingfenxingxing\\" style=\\"font-size: 26px; width: 26px; height: 26px; color: rgb(224, 224, 235);\\"></i></label>
  </div>
  <div class=\\"vvm-rate__icon\\" style=\\"width: 26px; height: 26px;\\">
    <transition-stub>
      <!--v-if-->
    </transition-stub><label class=\\"vvm-rate__defaultIcon\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconpingfenxingxing\\" style=\\"font-size: 26px; width: 26px; height: 26px; color: rgb(224, 224, 235);\\"></i></label>
  </div>
  <div class=\\"vvm-rate__icon\\" style=\\"width: 26px; height: 26px;\\">
    <transition-stub>
      <!--v-if-->
    </transition-stub><label class=\\"vvm-rate__defaultIcon\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconpingfenxingxing\\" style=\\"font-size: 26px; width: 26px; height: 26px; color: rgb(224, 224, 235);\\"></i></label>
  </div>
  <div class=\\"vvm-rate__icon\\" style=\\"width: 26px; height: 26px;\\">
    <transition-stub>
      <!--v-if-->
    </transition-stub><label class=\\"vvm-rate__defaultIcon\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconpingfenxingxing\\" style=\\"font-size: 26px; width: 26px; height: 26px; color: rgb(224, 224, 235);\\"></i></label>
  </div>
  <div class=\\"vvm-rate__icon\\" style=\\"width: 26px; height: 26px;\\">
    <transition-stub>
      <!--v-if-->
    </transition-stub><label class=\\"vvm-rate__defaultIcon\\"><i class=\\"vvm-iconfont vvm-icon vvm-ignore iconpingfenxingxing\\" style=\\"font-size: 26px; width: 26px; height: 26px; color: rgb(224, 224, 235);\\"></i></label>
  </div>
</div>"
`;
